core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java

```diff
@@ -198,16 +198,22 @@ public HiveSqlDialect(Context context) {
   }
 
   @Override public SqlOperator getTargetFunc(RexCall call) {
-    switch (call.type.getSqlTypeName()) {
-    case DATE:
-      switch (call.getOperands().get(1).getType().getSqlTypeName()) {
-      case INTERVAL_DAY:
-        if (call.op.kind == SqlKind.MINUS) {
-          return SqlLibraryOperators.DATE_SUB;
+    switch (call.getOperator().getKind()) {
+    case PLUS:
```

Comment:
This part shouldn't be here. We already have a case check for **PLUS** and **MINUS** in the calling method.